### ISSUE TITLE
[deno compile] - don't include lock file in compiled executable

### ISSUE THREAD

**Issue by cknight** [denoland/deno#21504]
At the moment, the deno.lock file is included in the deno compile executable.  This file seems unnecessary to have in the executable and would ideally be excluded.  [One user reported](https://discord.com/channels/684898665143206084/684898665151594506/1182435887037034528) a significantly large lock file adding some 75MB to the executable.

**Comment by dsherret** [comment]
I believe you're talking about npm modules? When compiling with a node_modules directory then this is impossible (because node_modules depend on other node_modules in mysterious and amazing ways), but we could trim down the dependencies when not using one.

**Comment by RuiNtD** [comment]
For clarification, the actual issue was with all modules in the lockfile being blindly baked into the executable, without actually considering if they're used. I tried to compile a hello world file (no dependencies) in the directory of an existing project and it created a ~140 MB EXE. As soon as I deleted the lockfile, the EXE dropped to ~82 MB. And there was no node_modules folder in the project.

**Comment by dsherret** [comment]
Yeah, it's just npm packages in the lockfile. `deno compile` should exclude npm packages when none are referenced in the graph and trim down unusued npm packages when not using a node_modules directory.

**Comment by ohroy** [comment]
any news ?

**Comment by notcome** [comment]
Can we remove/rename that lockfile temporarily during deno compile? It works on my side.

**Comment by ferm10n** [comment]
~~It seems like packages that used to be referenced in a deno.json and got an entry in deno.lock continue to exist in the lockfile even after removing the dependency from deno.json and the project entirely.~~

~~I think node package managers have the assumption that if a package is not referenced in the package.json or is a sub-dependency of something, it gets removed from the lock file~~

Edit: sorry I totally misunderstood what the issue is focusing on. I confused "deno.lock is included unnecessarily" with "unnecessary things are included in deno.lock"
